### ISSUE TITLE
Ruby 2.2 gives "LoadError: cannot load such file -- seven_zip_ruby/seven_zip_archive"

### ISSUE THREAD

**Issue by jjgh** [masamitsu-murase/seven_zip_ruby#17]
Windows 7, 64

> ruby -v
> ruby 2.2.2p95 (2015-04-13 revision 50295) [x64-mingw32]
> 
> gem query seven_zip_ruby
> **\* LOCAL GEMS ***
> seven_zip_ruby (1.2.4 x64-mingw32)

```
irb(main):001:0> require 'seven_zip_ruby'


LoadError: cannot load such file -- seven_zip_ruby/seven_zip_archive
        from C:/Dev/Ruby22-x64/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from C:/Dev/Ruby22-x64/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from C:/Dev/Ruby22-x64/lib/ruby/gems/2.2.0/gems/seven_zip_ruby-1.2.4-x64-mingw32/lib/seven_zip_ruby.rb:16:in `rescue in block in <top (required)>'
        from C:/Dev/Ruby22-x64/lib/ruby/gems/2.2.0/gems/seven_zip_ruby-1.2.4-x64-mingw32/lib/seven_zip_ruby.rb:12:in `block in <top (required)>'
        from C:/Dev/Ruby22-x64/lib/ruby/gems/2.2.0/gems/seven_zip_ruby-1.2.4-x64-mingw32/lib/seven_zip_ruby.rb:11:in `chdir'
        from C:/Dev/Ruby22-x64/lib/ruby/gems/2.2.0/gems/seven_zip_ruby-1.2.4-x64-mingw32/lib/seven_zip_ruby.rb:11:in `<top (required)>'
        from C:/Dev/Ruby22-x64/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:128:in `require'
        from C:/Dev/Ruby22-x64/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:128:in `rescue in require'
        from C:/Dev/Ruby22-x64/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:39:in `require'
        from (irb):1
        from C:/Dev/Ruby22-x64/bin/irb:11:in `<main>'
```

**Comment by caveman-dick** [comment]
I'm getting this same issue with my install:
```
ruby 2.2.3p173 (2015-08-18 revision 51636) [i386-mingw32]
```
@jjgh Did you manage to resolve your issue?

**Comment by jjgh** [comment]
@caveman-dick no I'm sorry, and I didn't have the time to investigate. I think what I did was just calling some command line 7z tool with `system`.

**Comment by caveman-dick** [comment]
@jjgh thanks I've managed to fix it myself by rolling back to v2.1 of ruby. For some reason the compilation of the gem didn't build the seven_zip_archive.so for 2.2. No ideas why though!

**Comment by orgads** [comment]
I rebuilt the library and it fixed it. You need to add a compilation for 2.2.

**Comment by Callisto82** [comment]
@orgads Can you please elaborate on how you solved the problem? How do I "add a compilation for 2.2"? Thanks :)

**Comment by caveman-dick** [comment]
:+1: I'm still a bit of a newbie to ruby so didn't know where to go once I found the issue! :)

**Comment by masamitsu-murase** [comment]
I'm so sorry for my late response.
I don't have enough time now, but I'll fix this issue soon.

The following is the cause of this issue.
* This gem includes pre-compiled binaries for Windows platform.  
  The current version of this gem has the binaries for both ruby2.0 and ruby2.1, but not for ruby2.2 and later.
* If you use rubyinstaller with devkit, you can install this gem as follows:  
    gem install --platform=ruby seven_zip_ruby

I'm sorry again.
Murase

**Comment by masamitsu-murase** [comment]
I released v1.2.5, which includes binaries for Ruby 2.2 and 2.3.
